File: WebApp/src/interfaces/song.ts
import { Genres } from "../models/genres";

export  interface ISong{
    id: number;
    name: string;
    band: string;
    genre: Genres
    state: State;
    url: string;
}

export interface State{
    location: Location;
    weather: Weather;
    mood: Mood;
}

export enum Location{
    Undefined,
    Transport,
    Home,
    Work,
    Nature
}

export enum Weather{
    Undefined,
    Rain,
    Sunny,
    Snow,
    Cloudy,
    ThunderStorm,
    Drizzle
}

export enum Mood{
    Undefined,
    Sad,
    Glad,
    Inspired,
    Enamored,
}

File: Model1/SongModel/Song.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Model
{
    public class Song
    {
        public Song()
        {
            PlaylistSongs = new List<PlaylistSong>();
        }

        public int ID { get; set; }

        public string Name { get; set; }

        public string Band { get; set; }

        public Genre Genre { get; set; }

        public State State { get; set; }

        public string Url { get; set; }

        public List<PlaylistSong> PlaylistSongs { get; set; }
    }
}
